pandas/tests/io/parser/na_values.py

```diff
@@ -224,6 +224,47 @@ def test_na_values_keep_default(self):
                                   'seven']})
         tm.assert_frame_equal(xp.reindex(columns=df.columns), df)
 
+        # see gh-19227: keep_default_na=False should be enforced
```

Comment:
can you split this test up, getting a bit long

Reply:
Fixed.